### ISSUE TITLE
Server-side printing continued

### ISSUE THREAD

**Issue by juanvallejo** [kubernetes/kubernetes#60712]
This is continuation of the work started in https://github.com/kubernetes/kubernetes/issues/58536. The below list outlines work items related to server-side printing in 1.11:

**Server**:
- [x] ~~determine whether to have a shorthand mime-type for table (application/json+meta+v1+Table) or similar~~ 
- [x] Move the OpenAPI column rendering for CRD to the server in the CRD implementation (continuation of #60269) https://github.com/kubernetes/kubernetes/pull/60991
- [x] Do a final pass on the default metadata renderer (the generic one)
- [x] Add a "example fuzzer" that returns a Table filled with all the values we could anticipate so that it's easy to test client impls
- [] ~~Review the Table API with a JS web console user~~ The current history proved the table output is very good. 
- [x] Look at a CSV mime type that uses table printing
- [x] Reevaluate the `creationTimestamp` to be returned as timestamp and not interpreted value.

**Client**:
- [x] Sorting and printing should be using the table but with full objects returned via the print API `(?includeObject=Object)` and then doing the sort on nested, as should custom columns.
- [x] Sorting in general needs to be looked at
- [x] Turn server-side print on by default (#61477)
- [x] Improve sorting in `kubectl get` (https://github.com/kubernetes/kubernetes/pull/67211 updated the sorter in `kubectl get` to handle server-side printing. We need to improve performance of the sorter to handle large lists of resources better)
- [ ] ~~Custom-columns in `kubectl get` need to support `Table` objects as part of server-side printing. Currently they fallback to old printing behavior~~ Custom columns will be using regular get since they need to operate on full object to be able to present fields requested by user. Using server-side print would be an overkill for this. 
- [x] `kubectl get --watch` behavior needs to be updated to use server-side printing. We need server events to include `Table` objects before we can move forward with this. - https://github.com/kubernetes/kubernetes/pull/76161

**Comment by juanvallejo** [comment]
@soltysh not sure if I can set the 1.11 milestone on this one.
@smarterclayton This tracks the items left over from https://github.com/kubernetes/kubernetes/issues/58536 that will be worked on for 1.11. We can close that issue in favor of this one.

**Comment by nikhita** [comment]
Repeating the sig label for the bot to pick it up and remove `needs-sig`.
/sig cli

**Comment by dims** [comment]
fun with bots!. let me try that

/sig cli

**Comment by BenTheElder** [comment]
that worked 🙃

**Comment by soltysh** [comment]
@juanvallejo I've added the task about re-evaulating the `creationTimestamp` to the task list. What we've discussed yesterday, which might help to improve UX especially when majority of sort operation are using that field and the current value we return is not much usable in that area.

**Comment by soltysh** [comment]
@juanvallejo we need to verify if the main bits here got in and move the remaining to 1.12

**Comment by jberkus** [comment]
So the only thing getting done on this for 1.11 is server-side-printing by default?  Is that correct?

**Comment by juanvallejo** [comment]
@jberkus 
> So the only thing getting done on this for 1.11 is server-side-printing by default? Is that correct?

This as well as the second item from the list "Move the OpenAPI column rendering for CRD to the server in the CRD implementation" were completed for 1.11

**Comment by jberkus** [comment]
OK, those are both merged, so should we change the milestone on this issue?

**Comment by soltysh** [comment]
@jberkus updated milestone.

**Comment by dims** [comment]
Looks like feature moved from 1.12 to 1.13 ( https://github.com/kubernetes/features/issues/515 ) Moving this issue as well. please move it back if needed.

/milestone v1.13

**Comment by shubheksha** [comment]
ping @juanvallejo, @soltysh - is this WIP for 1.13?

**Comment by juanvallejo** [comment]
@shubheksha I believe we're tracking any remaining client-side work for this feature in https://github.com/kubernetes/enhancements/issues/515.

@smarterclayton Do we have any plans to work on any of the server-side items for 1.13?

**Comment by smarterclayton** [comment]
No, there’s no urgency either.



On Oct 24, 2018, at 6:07 PM, Juan Vallejo <notifications@github.com> wrote:

@shubheksha <https://github.com/shubheksha> I believe we're tracking any
remaining client-side work for this feature in kubernetes/enhancements#515
<https://github.com/kubernetes/enhancements/issues/515>.

@smarterclayton <https://github.com/smarterclayton> Do we have any plans to
work on any of the server-side items for 1.13?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/issues/60712#issuecomment-432844802>,
or mute the thread

**Comment by shubheksha** [comment]
Thank you for the clarification!
Can we remove the milestone from this issue?

**Comment by nikopen** [comment]
as this is an umbrella issue with various non-urgent fixes/improvements, this can be tagged for v1.14

/priority important-longterm
/milestone v1.14

**Comment by soltysh** [comment]
I'm closing this in favor of https://github.com/kubernetes/enhancements/issues/515 and the work its tracking.
/close
